src/edu/stanford/nlp/sempre/FormulaEvaluator.java

```diff
@@ -0,0 +1,10 @@
+package edu.stanford.nlp.sempre;
+
+/**
+ * Given an Example containing multiple target formulas
+ * and a predicted formula, return a compatibility.
+ */
+public interface FormulaEvaluator extends ValueEvaluator {
```

Comment:
I think having ValueEvaluator and FormulaEvaluator (being a subclass) is really confusing.  I think this all should be consolidated into one interface called Evaluator with one method getCompatibility(Example ex, Derivation deriv).  Everyone will just implement this one method.